modules/Material/ActionBar.qml

```diff
@@ -203,6 +204,43 @@ Item {
             overflowMenu.close();
     }
 
+    /*!
+       \internal
+
+       put mousearea under everything blow
+     */    
+    MouseArea { 
+        anchors.fill: parent
+
+        property var previousPosition
+		property var enable : Storage.target.clientSideDecorations 
+
+        onPressed: {
+            previousPosition = Qt.point(mouseX, mouseY)
+        }
+
+        onDoubleClicked: {
+			if( enable ){
+				if( Storage.target.visibility != Window.FullScreen ){
+					Storage.target.visibility = Window.FullScreen
+				}else{
+					Storage.target.visibility = Window.Windowed
+				}
+			}
+        }
+
+        onPositionChanged: {
+            if (pressedButtons == Qt.LeftButton 
+				&& Storage.target.visibility != Window.FullScreen 
+				&& enable ) {
+                var dx = mouseX - previousPosition.x
+                var dy = mouseY - previousPosition.y
+                Storage.target.x = Storage.target.x + dx 
+                Storage.target.y = Storage.target.y + dy
```

Comment:
It is a race condition, works fine on windows for some reason.  It can be fixed by exposing QCursor::pos() from c++ to get the screen position of the mouse.